kqueen_ui/generic_views.py

```diff
@@ -40,7 +44,10 @@ def _handle_response(self, response, resource, action):
                 user_msg = '{} not found.'.format(fmt_resource)
                 self.graceful_exit(msg, user_msg)
             elif response.status >= 400:
-                user_msg = 'Error occurred while contacting backend, please try again later.'
+                if response.error:
+                    user_msg = response.error
+                else:
+                    user_msg = 'Error occurred while contacting backend, please try again later.'
```

Comment:
i prefer to move response.error as single case and remove redundant nested dependencies
```
        if response.error:
            user_msg = response.error
            self.graceful_exit(msg, user_msg)
        elif response.status == -1:
            user_msg = 'Backend is unavailable at this time, please try again later.'
            self.graceful_exit(msg, user_msg)
            self.graceful_exit(msg, user_msg)
        elif response.status == 401:
            fmt_action = str(action).lower()
            fmt_resource = str(resource).lower()
            user_msg = 'You are not authorized to {} {}.'.format(fmt_action, fmt_resource)
            self.graceful_exit(msg, user_msg)
        elif response.status == 404:
            fmt_resource = str(resource).capitalize()
            user_msg = '{} not found.'.format(fmt_resource)
            self.graceful_exit(msg, user_msg)
        elif response.status >= 400:
            user_msg = 'Error occurred while contacting backend, please try again later.'
            self.graceful_exit(msg, user_msg)
        return response.data